lib/facter/gid_max.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+Facter.add(:gid_max) do
+  setcode do
+    lines = File.readlines("/etc/login.defs")
+    lines.find { |line| line.start_with?("GID_MAX") }.split("\s")[1].strip
```

Comment:
What happens if the line looks like this:

```
GID_MAX        50000
```

That is if there are multiple spaces between GID_MAX and the value? Will \[1\] resolve as " " (=a space)?

Also, what if there are multiple lines that start with GID_MAX? Normally that would not happen, but can't be ruled out. That would cause potentially hard to debug problems. Two options I can think of for handling that:

* Error out with a reasonable error message ("ERROR: multiple GID_MAX lines found in /etc/login.defs")
* Pick the first value (e.g. *line.starts_with?("GID_MAX")[0]*) and print an informal message ("WARNING: multiple GIT_MAX lines found in /etc/login.defs").

Reply:
Based on testing this works as expected. The "find" method returns the first line that matches. The split("\s") considers multiple whitespace characters as one, so multiple spaces between GID_MAX and the value do not cause issues.